src/jit/codegenxarch.cpp

```diff
@@ -6640,34 +6640,7 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
             ins = INS_mov;
         }
 
-        if (ins == INS_AND)
-        {
-            noway_assert(isUnsignedDst);
-
-            /* Generate "and reg, MASK */
-            unsigned fillPattern;
-            if (dstSize == EA_1BYTE)
-            {
-                fillPattern = 0xff;
-            }
-            else if (dstSize == EA_2BYTE)
-            {
-                fillPattern = 0xffff;
-            }
-            else
-            {
-                fillPattern = 0xffffffff;
-            }
-
-            inst_RV_IV(INS_AND, targetReg, fillPattern, EA_4BYTE);
-        }
-#ifdef _TARGET_AMD64_
-        else if (ins == INS_movsxd)
-        {
-            inst_RV_RV(ins, targetReg, sourceReg, srcType, srcSize);
-        }
-#endif // _TARGET_AMD64_
-        else if (ins == INS_mov)
+        if (ins == INS_mov)
```

Comment:
I presume in the past we use the "and" instruction to perform  a zero extending casts.
So could you also add an assert here that we don't have an and:
`assert(ins != INS_AND);`